homeassistant/components/tradfri/cover.py

```diff
@@ -26,9 +26,11 @@ async def async_setup_entry(
     api = tradfri_data[KEY_API]
     devices = tradfri_data[DEVICES]
 
-    covers = [dev for dev in devices if dev.has_blind_control]
-    if covers:
-        async_add_entities(TradfriCover(cover, api, gateway_id) for cover in covers)
+    entities = [
```

Comment:
To really simplify, consider just always calling `async_add_entities` on the list. IMO if we think this is a big performance hit, we can add a check inside of async_add_entities that makes an empty list a no-op so that all integrations don't need to bother with this check.

Reply:
Currently I cannot just call it, because it does not allow an empty array.

Personally I find that it should allow an empty array, of course not for the performance, but for the readability.

Reply:
Furthermore I see the change as easier readable (I did not do it for performance)

Reply:
My apologies !!! my knowledge of async_add_entities was old. 

I have implemented your suggestion.